crates/top/re_sdk/src/lenses/ast.rs

```diff
@@ -234,70 +301,131 @@ impl LensBuilder {
                 entity_path_filter,
                 component: component.into(),
             },
-            outputs: vec![],
+            component_columns: vec![],
+            time_columns: vec![],
         })
     }
 
     /// Can be used to define one or more output columns that are derived from the
     /// component specified via [`Self::for_input_column`].
     ///
-    /// The output column will live on the same entity path as the input column.
+    /// The new column will be placed on the same entity path as the input column.
     pub fn add_output_column(
         mut self,
         component_descr: ComponentDescriptor,
         ops: impl IntoIterator<Item = Op>,
     ) -> Self {
-        let column = OutputColumn {
+        let column = ComponentColumn {
             new_entity_path: None,
             component_descr,
             ops: ops.into_iter().collect(),
             is_static: false,
         };
-        self.0.outputs.push(column);
+        self.0.component_columns.push(column);
         self
     }
 
     /// Can be used to define one or more output columns that are derived from the
     /// component specified via [`Self::for_input_column`].
     ///
-    /// The output column will live on the same entity path as the input column.
+    /// The new column will be placed at `entity_path`.
     pub fn add_output_column_entity(
         mut self,
         entity_path: impl Into<EntityPath>,
         component_descr: ComponentDescriptor,
         ops: impl IntoIterator<Item = Op>,
     ) -> Self {
-        let column = OutputColumn {
+        let column = ComponentColumn {
             new_entity_path: Some(entity_path.into()),
             component_descr,
             ops: ops.into_iter().collect(),
             is_static: false,
         };
-        self.0.outputs.push(column);
+        self.0.component_columns.push(column);
+        self
+    }
+
+    /// Can be used to extract a time column from the component specified via
+    /// [`Self::for_input_column`].
+    ///
+    /// The output column will be placed on the same entity path as the input column.
```

Comment:
something is missing here: I was confused about whether this produces a time column on the output or is about input etc.; looking holistically at the API that got clear quickly, but maybe we can do a bit better:

on the output entity columns we call out that it's an output, so we should do the same here? 
-> `add_time_output_column`/ `add_output_time_column`?